src/ipc/ipc4/handler.c

```diff
@@ -914,6 +914,99 @@ static int ipc4_unbind_module_instance(struct ipc4_message_request *ipc4)
 	return ipc_comp_disconnect(ipc, (ipc_pipe_comp_connect *)&bu);
 }
 
+static int  ipc4_get_vendor_config_module_instance(struct comp_dev *dev,
+						   const struct comp_driver *drv,
+						   uint32_t module_id,
+						   uint32_t instance_id,
+						   bool init_block,
+						   bool final_block,
+						   uint32_t *data_off_size,
+						   char *data_out,
+						   const char *data_in)
+{
+	const struct sof_tl *input_tl = (struct sof_tl *)data_in;
+	int ret;
+	struct ipc4_vendor_error *error;
+
+	if (init_block && final_block) {
+		/* we use data_off_size as in/out,
+		 *  save value to new variable so it can be used as out size
+		 */
+		const uint8_t *end_offset = data_in + *data_off_size;
```

Comment:
```
/home/runner/work/sof/sof/src/ipc/ipc4/handler.c:930:45:
 error: pointer targets in initialization of ‘const uint8_t *’
 {aka ‘const unsigned char *’} from ‘const char *’ differ in signedness [-Werror=pointer-sign]
  930 |                 const uint8_t *end_offset = data_in + *data_off_size;
      |                                             ^~~~~~~
```

Reply:
Yeah, this looks better. I would still move '*data_off_size/sizeof(struct sof_tl)' to a const var, so that we don't divide each iteration.